VECTOR.md

```
@@ -0,0 +1,23 @@
+# Vector
+
+Let's use Pinecone for the following:
+
+1. As we use the AI SDK, we will use OpenAI compatible models for embedding, like embedding large (use models, embedding in models.ts), same with searching for data.
+2. We use a dense index in Pinecone for semantic search, matching the output of modern embedding models and the bot's use case. (https://docs.pinecone.io/guides/index-data/create-an-index#bring-your-own-vectors), include metadata like timestamp, user, server, channel, etc. So when searching we can limit the search to a specific server, channel, etc. (https://docs.pinecone.io/guides/search/filter-by-metadata#search-with-a-metadata-filter)
```

markdownlint-cli2 (0.17.2)

Bare URL used

(MD034, no-bare-urls)